docs/deprecation-policy.md

```diff
@@ -0,0 +1,186 @@
+no less than:**
+   * **GA: 1 year or 2 releases (whichever is longer)**
+   * **Beta: 3 months or 1 release (whichever is longer)**
+   * **Alpha: 0 releases**
+
+This is best illustrated by example.  Imagine a Kubernetes release, version X,
+which supports a particular API group.  A new Kubernetes release is made every
+approximately 3 months (4 per year).  The following table describes which API
+versions are supported in a series of subsequent releases.
+
+| Release | API versions | Notes |
+|---------|--------------|-------|
+| X       | v1           |       |
+| X+1     | v1, v2alpha1 |       |
+| X+2     | v1, v2alpha2 | * v2alpha1 is removed, “action required” relnote |
+| X+3     | v1, v2beta1  | * v2alpha2 is removed, “action required” relnote |
+| X+4     | v1, v2beta1, v2beta2 | * v2beta1 is deprecated, “action required” relnote |
+| X+5     | v1, v2, v2beta2      | * v2beta1 is removed, “action required” relnote<br> * v2beta2 is deprecated, “action required” relnote<br> * v1 is deprecated, “action required” relnote |
+| X+6     | v1, v2               | * v2beta2 is removed, “action required” relnote |
+| X+7     | v1, v2               | |
+| X+8     | v1, v2               | |
+| X+9     | v2                   | * v1 is removed, “action required” relnote |
+
+### REST resources (aka API objects)
+
+Consider a hypothetical REST resource named Widget, which was present in API v1
+in the above timeline, and which we want to deprecate.  We [document](http://kubernetes.io/docs/deprecated/)
+and [announce](https://groups.google.com/forum/#!forum/kubernetes-announce) the
+deprecation in sync with release X+1.  The Widget resource still exists in API
+version v1 (deprecated) but not in v2alpha1.  The Widget resource continues to
+exist and function in releases up to and including X+8.  Only in release X+9,
+when API v1 has aged out, does the Widget resource cease to exist, and the
+behavior get removed.
+
+### Fields of REST resources
+
+As with whole REST resources, an individual field which was present in API v1
+must exist and function until API v1 is removed.  Unlike whole resources, the
+v2 APIs may choose a different representation for the field, as long as it can
+be round-tripped.  For example a v1 field named “magnitude” which was
+deprecated might be named “deprecatedMagnitude” in API v2.  When v1 is
+eventually removed, the deprecated field can be removed from v2.
+
+### Enumerated or constant values
+
+As with whole REST resources and fields thereof, a constant value which was
+supported in API v1 must exist and function until API v1 is removed.
+
+### Component config structures
+
+Component configs are versioned and managed just like REST resources.
+
+### Future work
+
+We intend to introduce more fine-grained API versions, at which point we will
+adjust these rules as needed.
+
+## Deprecating a flag or CLI
+
+The Kubernetes system is comprised of several different programs cooperating.
```

Comment:
"we" construct.

"Sometimes, a Kubernetes release might remove flags or CLI commands"